apps/src/templates/lessonOverview/LessonOverview.jsx

```diff
@@ -62,14 +70,15 @@ class LessonOverview extends Component {
     );
   }
 
-  recordAndNavigateToPdf = (e, firehoseKey, url) => {
-    // Prevent navigation to url until callback
-    e.preventDefault();
+  recordAndHandleResource = (e, firehoseKey, action, url = null) => {
+    e.preventDefault(); // Prevent navigation to url until callback
+    const event =
+      action === ResourceActions.NAVIGATE ? 'open-pdf' : 'print-from-browser';
```

Comment:
I keep `'open-pdf'` and `'print-from-browser'` strings here, instead of declaring them at the beginning, since they are exclusively used in `firehoseClient.putRecord`.